lib/utils/job-utils/racadm-tool.js

```diff
@@ -192,29 +220,136 @@ function racadmFactory(
         }
         if (fileInfo.style === 'remote') {
             command = "update -f " + fileInfo.name + " -u " +
                 cifsUser + " -p " + cifsPassword + " -l " + fileInfo.path;
-        } else if (fileInfo.style === 'local') {
+        } else { // 'local'
             command = "update -f " + fileInfo.path + "/" + fileInfo.name;
-        } else {
-            throw new Error('iDRAC image file path is invalid');
         }
+        return self.runCommand(host, user, password, command, 0, 1000)
+            .then(function(){
+                return self.runCommand(host, user, password, "serveraction powercycle", 0, 1000);
+            })
+            .then(function(){
+                return self.getLatestJobId(host, user, password);
+            })
+            .then(function(jobId){
+                return self.waitJobDone(host, user, password, jobId, 0, 1000);
+            });
+    };
+
+    /**
+     * Get idrac software inventory
+     *
+     * @param {string} host
+     * @param {string} user
+     * @param {string} password
+     * @return {object} - software inventory
+     */
+    RacadmTool.prototype.getSoftwareList = function(host, user, password) {
+        var command = "swinventory";
+        return this.runCommand(host, user, password, command)
+            .then(function(stdout){
+                return parser.getSoftwareList(stdout);
+            });
+    };
+
+    /**
+     * Get BIOS xml file and store it to desired file path and file name
+     *
+     * @param {string} host
+     * @param {string} user
+     * @param {string} password
+     * @param {Object} cifsConfig -  includes samba password, username and bios.xml file path
+     * Password and username are only required when fiel path is a cifs share folder.
+     * @param {string} device - Dell device Full-Qualified-Device-Descriptor (FQDD)
+     * @returns {object} - job status
+     */
+    RacadmTool.prototype.getBiosConfig = function(host, user, password, cifsConfig, device) {
+        var self =  this;
+        var FILEPATH = '/tmp/configure.xml'; //root authority required
+        var command = '', fqdd, cifsUser, cifsPassword, fileStoredPath;
+        cifsConfig = cifsConfig || {};
+        cifsUser = cifsConfig.user || '';
+        cifsPassword = cifsConfig.password || '';
+        fileStoredPath = cifsConfig.filePath || FILEPATH;
+
+        return Promise.resolve()
+            .then(function(){
+                return self.getSoftwareList(host, user, password);
+            })
+            .then(function(softwareList){
+                if (typeof softwareList.BIOS.FQDD === "undefined"){
+                    throw new Error ('Can not get BIOS FQDD');
+                }
+                fqdd = device || softwareList.BIOS.FQDD;
+            })
+            .then(function(){
+                return parser.getPathFilename(fileStoredPath);
+            })
+            .then(function(fileInfo){
```

Comment:
This design is OK, so following is just a suggestion, it may be more beautiful and most importantly we can avoid the the bridged variable`fqdd` outside the scope:
```javascript
.then( function()
xxx
var fqdd = device || softwareList.BIOS.FQDD;
var fileInfo parser.getPathFilename(fileStoredPath);
return [fqdd, fileInfo];
}).spread(fqdd, fileInfo) {
 xxx
}
```